bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -174,13 +181,20 @@ impl Config {
         }
     }
 
-    /// Enables host registration WasmEdge process.
-    pub fn enable_wasmedge_process(self) -> Self {
+    /// Enables or disables host registration WasmEdge process.
+    pub fn wasmedge_process(self, enable: bool) -> Self {
         unsafe {
-            wasmedge::WasmEdge_ConfigureAddHostRegistration(
-                self.ctx,
-                wasmedge::WasmEdge_HostRegistration_WasmEdge_Process,
-            )
+            if enable && !self.wasmedge_process_enabled() {
```

Comment:
In my option, if I call `wasmedge_precess(enable=true)`, then it must be set to enable.
Why do we need to check the `wasmedge_process_enabled`?

Reply:
The predicate is to say that the `wasmedge_process` option turns on if and only if `enable` is `true` and the `wasmedge_process` option is off currently. If the `wasmedge_process` option was on, no need to turn it on again.

I'm not sure if the relevant C API performs the check or not, so I put it there. If it's not necessary, I can simplify the design. Thanks a lot!

Reply:
This API will call `Conf.addHostRegistration(static_cast<WasmEdge::HostRegistration>(Host));` [1].
Then, it will call `Hosts.set(static_cast<uint8_t>(Host));` [2].

And this API will not check the previous state, it will enable/disable the configure option directly.

[1]: https://github.com/WasmEdge/WasmEdge/blob/master/lib/api/wasmedge.cpp#L644-L651
[2]: https://github.com/WasmEdge/WasmEdge/blob/master/include/common/configure.h#L149-L151

Reply:
The original design has been refactored, please review the latest changes again. Thanks!